packages/lesswrong/lib/utils/viewUtils.ts

```diff
@@ -167,23 +167,32 @@ export const jsonArrayContainsSelector = <T extends DbObject>(
   ? {$expr: {$jsonArrayContains: [field, value]}}
   : {[field]: value};
 
+const removeAndOr = <T extends DbObject>(selector: MongoSelector<T>) => {
+  const copy = {...selector}
+  delete copy.$and
+  delete copy.$or
+  return copy
+}
+
 const mergeTwoSelectors = <T extends DbObject>(
   baseSelector?: MongoSelector<T>,
   newSelector?: MongoSelector<T>,
 ) => {
   if (!baseSelector) return newSelector;
   if (!newSelector) return baseSelector;
-  let mergedSelector = merge(baseSelector, newSelector)
-  if ("$and" in baseSelector && "$and" in newSelector) {
+
+  // Remove $ands and $ors before merging, and then add them back in with the correct logic
+  let mergedSelector = merge(removeAndOr(baseSelector), removeAndOr(newSelector));
```

Comment:
There was a bug here caused by using lodash's `merge` without removing `$and` and `$or` first